control-plane/agents/src/bin/ha/node/server.rs

```diff
@@ -262,6 +268,36 @@ impl NodeAgentOperations for NodeAgentSvc {
         };
         Ok(())
     }
+
+    async fn get_nvme_controller(
+        &self,
+        request: &dyn GetControllerInfo,
+        _context: Option<Context>,
+    ) -> Result<NvmeSubsys, ReplyError> {
+        let uri = request
+            .nvme_path()
+            .parse::<Uri>()
+            .map_err(|_| SvcError::InvalidArguments {})?;
+        let nqn = uri.path()[1 ..].to_string();
+        match Subsystem::try_from_nqn(nqn.as_str()) {
+            Ok(subsys_list) => {
+                let mut controller_list = vec![];
+                for controller in subsys_list {
+                    if controller.state == "live" && controller.state == "connecting" {
+                        controller_list.push(NvmeCtrller::new(controller.clone().address))
+                    } else {
+                        continue;
+                    }
+                }
```

Comment:
nit: can replace this with something like:
```suggestion
                let controller_list = subsys_list
                                  .into_iter()
                                  .filter(|controller| controller.state == "live" && controller.state == "connecting")
                                  .map(|controller| NvmeCtrller::new(controller.address))
                                  .collect();
```

Reply:
Thanks. Didnt know how to filter in map's closure. Incorporated